NPE in AuthenticatorActivity.onActivityResult

via GPlay:
```
java.lang.RuntimeException: 
  at android.app.ActivityThread.deliverResults (ActivityThread.java:5041)
  at android.app.ActivityThread.handleSendResult (ActivityThread.java:5084)
  at android.app.ActivityThread.-wrap20 (Unknown Source)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2053)
  at android.os.Handler.dispatchMessage (Handler.java:108)
  at android.os.Looper.loop (Looper.java:166)
  at android.app.ActivityThread.main (ActivityThread.java:7529)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:245)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:921)
Caused by: java.lang.NullPointerException: 
  at com.owncloud.android.authentication.AuthenticatorActivity.onActivityResult (AuthenticatorActivity.java:2418)
  at android.app.Activity.dispatchActivityResult (Activity.java:7701)
  at android.app.ActivityThread.deliverResults (ActivityThread.java:5037)
```

It seems there are cases where the scan probably crashed and returns while the result won't have the string we are looking for, so we need to mae this null-safe.

--- a/src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java
+++ b/src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java
@@ -2415,7 +2415,7 @@ public class AuthenticatorActivity extends AccountAuthenticatorActivity
 
             String result = data.getStringExtra("com.blikoon.qrcodescanner.got_qr_scan_relult");
 
-            if (!result.startsWith(getString(R.string.login_data_own_scheme))) {
+            if (result == null || !result.startsWith(getString(R.string.login_data_own_scheme))) {
                 mServerStatusIcon = R.drawable.ic_alert;
                 mServerStatusText = "QR Code could not be read!";
                 showServerStatus();

